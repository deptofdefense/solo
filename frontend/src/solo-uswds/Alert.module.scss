.extWidth {
    width: 100vw;
}