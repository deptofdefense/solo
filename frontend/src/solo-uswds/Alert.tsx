import React from "react";
import classNames from "classnames";

interface AlertProps extends Partial<JSX.IntrinsicElements["div"]> {
  status: "success" | "warning" | "error" | "info";
  heading?: string;
  text?: string;
}

export const Alert: React.FC<AlertProps> = ({
  status,
  heading,
  text,
  className,
  ...rest
}) => (
  <div
    className={classNames("usa-alert", `usa-alert--${status}`, className)}
    {...rest}
  >
    <div className="usa-alert__body">
      {heading && <h3 className="usa-alert__heading">{heading}</h3>}
      {text && <p className="usa-alert__text">{text}</p>}
    </div>
  </div>
);
