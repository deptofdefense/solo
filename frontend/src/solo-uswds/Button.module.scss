.square {
  border-radius: 0 !important;
}
