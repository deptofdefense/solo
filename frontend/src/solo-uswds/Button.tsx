import React, { useState } from "react";
import classNames from "classnames";

interface ButtonProps {
  focused?: boolean;
  active?: boolean;
  disabled?: boolean;
  outline?: boolean;
  inverse?: boolean;
  big?: boolean;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  color?: "secondary" | "accent-cool" | "base";
  className?: string;
}

export const Button: React.FC<ButtonProps> = ({
  children,
  onClick,
  color,
  focused = false,
  active = false,
  outline = false,
  inverse = false,
  big = false,
  disabled = false,
  className = ""
}) => {
  const [isHovered, setIsHovered] = useState(false);

  return (
    <button
      onClick={onClick}
      onMouseEnter={() => setIsHovered(true)}
      onMouseLeave={() => setIsHovered(false)}
      disabled={disabled}
      className={classNames(className, "usa-button", {
        "usa-button--hover": isHovered,
        "usa-button--active": active,
        "usa-button--outline": outline,
        "usa-button--inverse": inverse,
        "usa-focus": focused,
        "usa-button--secondary": color === "secondary",
        "usa-button--accent-cool": color === "accent-cool",
        "usa-button--base": color === "base",
        "usa-button--big": big
      })}
    >
      {children}
    </button>
  );
};
