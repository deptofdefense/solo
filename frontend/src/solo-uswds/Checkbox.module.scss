.inp {
  position: unset !important;
  left: 0 !important;
}
