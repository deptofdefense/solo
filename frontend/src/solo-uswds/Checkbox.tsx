import React from "react";
import classNames from "classnames";
import classes from "./Checkbox.module.scss";

interface CheckboxProps extends Partial<JSX.IntrinsicElements["input"]> {
  checked?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
}

export const Checkbox: React.FC<CheckboxProps> = ({
  checked,
  onChange,
  className,
  ...rest
}) => (
  <input
    className={classNames("usa-checkbox__input", classes.inp, className)}
    onChange={onChange}
    type="checkbox"
    checked={checked}
    {...rest}
  />
);
