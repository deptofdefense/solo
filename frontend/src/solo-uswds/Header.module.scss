.navContainer {
  max-width: none !important;
  margin-top: 2rem;
  align-items: flex-start !important;
}
@media only screen and (max-width: 64em) {
  div.navContainer {
    margin-top: 0rem;
  }
}