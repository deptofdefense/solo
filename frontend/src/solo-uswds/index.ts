export { Button } from "./Button";
export { OfficialSiteBanner } from "./OfficialSiteBanner";
export { Header, HeaderNavbar } from "./Header";
export { HeaderNav } from "./HeaderNav";
export { HeaderNavLink } from "./HeaderNavLink";
export { HeaderLogo } from "./HeaderLogo";
