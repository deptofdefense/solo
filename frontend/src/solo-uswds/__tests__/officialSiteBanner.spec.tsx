import React from "react";
import { OfficialSiteBanner } from "solo-uswds";
import { render, fireEvent, wait } from "@testing-library/react";

describe("Official site banner component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<OfficialSiteBanner />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("starts off closed", async () => {
    const { queryByText } = render(<OfficialSiteBanner />);
    expect(queryByText(/site is secure./)).not.toBeInTheDocument();
    expect(queryByText(/.mil means it's official./)).not.toBeInTheDocument();
  });

  it("toggles here's how you know click", async () => {
    const { getByText, queryByText } = render(<OfficialSiteBanner />);
    const toggleElem = getByText("Here's how you know");
    expect(toggleElem).toBeInTheDocument();
    fireEvent.click(toggleElem);
    await wait(() => {
      expect(getByText(/site is secure/)).toBeInTheDocument();
      expect(getByText(/mil means it's official./)).toBeInTheDocument();
    });
    fireEvent.click(toggleElem);
    await wait(() => {
      expect(queryByText(/site is secure/)).not.toBeInTheDocument();
      expect(queryByText(/mil means it's official./)).not.toBeInTheDocument();
    });
  });
});
