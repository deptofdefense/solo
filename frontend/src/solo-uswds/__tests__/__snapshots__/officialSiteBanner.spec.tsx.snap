// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Official site banner component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="usa-banner"
  >
    <header
      class="usa-banner__header"
    >
      <div
        class="usa-banner__inner"
      >
        <div
          class="grid-col-auto"
        >
          <img
            alt="U.S. flag"
            class="usa-banner__header-flag"
            src="us_flag_small.png"
          />
        </div>
        <div
          class="grid-col-fill tablet:grid-col-auto"
        >
          <p
            class="usa-banner__header-text"
          >
            An official website of the United States government
          </p>
        </div>
        <button
          aria-controls="gov-banner"
          aria-expanded="false"
          class="usa-accordion__button usa-banner__button"
        >
          <span
            class="usa-banner__button-text"
          >
            Here's how you know
          </span>
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
